src/main/java/frc/robot/subsystems/AlgaeBat.java

```diff
@@ -0,0 +1,122 @@
+        algaeBatConfig.softLimit.forwardSoftLimitEnabled(true);
+        algaeBatConfig.softLimit.reverseSoftLimitEnabled(true);
+        pivotMotor.configure(algaeBatConfig, ResetMode.kResetSafeParameters, PersistMode.kPersistParameters);
+    }
+
+    public void runpivotmotor(double speed) {
+        pivotMotor.set(speed);
+    }
+
+    public boolean batInKnockPosition() {
+        double pivotposition = getPivotAngle();
+        boolean pivotKnock = pivotposition >= AlgaeBatConstants.ALGAE_BAT_KNOCK_POSITION;
+
+        return pivotKnock;
+    }
+
+    public boolean batInHomePosition() {
+        double pivotposition = getPivotAngle();
+        boolean pivothome = pivotposition >= AlgaeBatConstants.ALGAE_BAT_HOME_POSITION
+                && pivotposition < AlgaeBatConstants.ALGAE_BAT_HOME_POSITION + 0.01;
+
+        return pivothome;
+    }
+
+    public double getPivotAngle() {
+        return pivotMotor.getEncoder().getPosition();
+    }
+
+    public void setBrake(boolean enabled, boolean sync) {
+        SparkMaxConfig config = new SparkMaxConfig();
+        if (enabled) {
+            config
+                    .idleMode(IdleMode.kBrake);
+        }
+
+        else {
+            config
+                    .idleMode(IdleMode.kCoast);
+        }
+
+        if (sync) {
+            pivotMotor.configure(config, ResetMode.kNoResetSafeParameters, PersistMode.kNoPersistParameters);
+        } else {
+            pivotMotor.configureAsync(config, ResetMode.kNoResetSafeParameters, PersistMode.kNoPersistParameters);
+        }
+    }
+
+    public Command extendBat(
+            AlgaeBat algaeBat) {
+        return Commands.startEnd(
+                () -> {
+                double extensionAngle = AlgaeBatConstants.ALGAE_BAT_KNOCK_POSITION - getPivotAngle();
+                double scalingFactor = extensionAngle / (AlgaeBatConstants.ALGAE_BAT_KNOCK_POSITION - AlgaeBatConstants.ALGAE_BAT_HOME_POSITION);
+                pivotMotor.set((AlgaeBatConstants.ALGAE_BAT_REACH_SPEED * scalingFactor) + 0.02);
+                SmartDashboard.putBoolean("extendBat", true);
+                SmartDashboard.putNumber("extendBatSpeed", AlgaeBatConstants.ALGAE_BAT_REACH_SPEED * scalingFactor);
+                },
+                () -> {pivotMotor.set(0);
+                SmartDashboard.putBoolean("extendBat", false);},
+                algaeBat)
+                .onlyWhile(
+                        () -> (!batInKnockPosition()));
+    }
+
+    public Command returnBat(
+            AlgaeBat algaeBat) {
+        return Commands.startEnd(
+                () -> {
+                    double extensionAngle = AlgaeBatConstants.ALGAE_BAT_HOME_POSITION - getPivotAngle();
+                    double scalingFactor = extensionAngle
+                            / (AlgaeBatConstants.ALGAE_BAT_HOME_POSITION - AlgaeBatConstants.ALGAE_BAT_KNOCK_POSITION);
+                    pivotMotor.set((AlgaeBatConstants.ALGAE_BAT_RETURN_SPEED * scalingFactor) + 0.02);
```

Comment:
Technically, if we wanted a constant bit of speed added here, I think we'd want to subtract here to have the correct directionality. But fine.